diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/WhenMustHaveElseWarnTest.kt

```diff
@@ -90,4 +90,38 @@ class WhenMustHaveElseWarnTest : LintTestBase(::WhenMustHaveElseRule) {
                 """.trimMargin()
         )
     }
+
+    @Test
+    @Tag(WarningNames.WHEN_WITHOUT_ELSE)
+    fun `when in func only enum entries`() {
+        lintMethod(
+            """
+                |fun foo() {
+                |    val v: Enum
+                |    when (v) {
+                |        Enum.ONE, Enum.TWO -> foo()
```

Comment:
What if enum members are imported (like `import Enum.*`) and used like `ONE, TWO`?

Reply:
It this case we should either check the uppercase (which is weird) or admit that those `ONE, TWO` could be enum members so that we should not put `else` into `when`.

Reply:
Yes, I think the latter is reasonable. If when entries are only dot-qualified or plain reference expressions, then we should ignore them.

Reply:
Do we need to put else in this when block?

```
val a: Int = 1
when(a) {
    1 -> foo()
}
```

Reply:
yes

Reply:
Ok, thanks